subt/artf_node.py

```diff
@@ -104,11 +106,13 @@ def detect(self, image):
         assert self.width == img.shape[1], (self.width, img.shape[1])
 
         result = self.detector(img)
-        if len(result) > 0:
-            self.stdout(result)
-            report = result2report(result, self.depth)
-            if report is not None:
-                self.publish('artf', report)
-                self.publish('debug_artf', image)  # JPEG
+        result_cv = self.cv_detector(img)
+        if len(result) > 0 and len(result_cv) > 0:
+            if result[0][0] == result_cv[0][0]: # check artefacts names
```

Comment:
This seems to assume that each of the detectors detected a single artifact. However, for example in Urban Virtual, there were rooms with multiple different artifacts in the view. Can you generalize this part?

Reply:
Yes, i know this limitation. However, this come from the original code where the result2report function ignores multiple object as well. I would prefer improve this thing in next step.